File: Coverage of all Zeros in a Binary Matrix.java
class Solution {
    public int findCoverage(int[][] matrix) {
        // code here
        int result = 0;
        
        for(int i = 0; i < matrix.length; i++) {
            for(int j = 0; j < matrix[i].length; j++) {
                if(matrix[i][j] == 0) {
                    // Left
                    if(((j - 1) >= 0) && matrix[i][j - 1] == 1) {
                        result++;    
                    }
                    
                    // right
                    if(((j + 1) < matrix[i].length) && matrix[i][j + 1] == 1) {
                        result++;    
                    }
                    
                    // up
                    if((((i - 1) >= 0) && matrix[i - 1].length > j) && matrix[i - 1][j] == 1) {
                        result++;
                    }
                    
                    // down
                    if((((i + 1) < matrix.length) && matrix[i + 1].length > j) && matrix[i + 1][j] == 1) {
                        result++;
                    }
                }
            }
        }
        return result;
    }
}

File: Coverage of all Zeros in a Binary Matrix.cpp
class Solution {
  public:
    int findCoverage(vector<vector<int>>& mat) {
        // code here
        int n = mat.size();
        int m = mat[0].size();
        
        int coverage = 0;
        
        // Left
        for(int i = 0; i < n; i++) {
            bool foundOne = false;
            for(int j = 0; j < m; j++) {
                if(mat[i][j] == 1) foundOne = true;
                else {
                    if(foundOne) coverage++;
                }
            }
        }
        // Right
        for(int i = 0; i < n; i++) {
            bool foundOne = false;
            for(int j = m-1; j >= 0; j--) {
                if(mat[i][j] == 1) foundOne = true;
                else {
                    if(foundOne) coverage++;
                }
            }
        }
        
        // top to bottom
        for(int i = 0; i < m; i++) {
            bool foundOne = false;
            for(int j = 0; j < n; j++) {
                if(mat[j][i] == 1) foundOne = true;
                else {
                    if(foundOne) coverage++;
                }
            }
        }
        
        
        for(int i = 0; i < m; i++) {
            bool foundOne = false;
            for(int j = n-1; j >= 0; j--) {
                if(mat[j][i] == 1) foundOne = true;
                else {
                    if(foundOne) coverage++;
                }
            }
        }
        
        return coverage;
    }
};
